extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/query/FieldFilter.java

```diff
@@ -0,0 +1,103 @@
+package mil.nga.giat.geowave.datastore.accumulo.query;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.index.StringUtils;
+import mil.nga.giat.geowave.core.store.dimension.NumericDimensionField;
+import mil.nga.giat.geowave.core.store.index.CommonIndexValue;
+
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.Filter;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+
+import com.google.common.base.Splitter;
+
+public class FieldFilter extends
+		Filter
+{
+	// must execute prior to WholeRowIterator/QueryFilterIterator but after
+	// SharedVisibilitySplittingIterator
```

Comment:
Do we have an issue where subset fields may not include fields used in the filter?